crates/task-impls/src/consensus/mod.rs

```diff
@@ -156,132 +334,13 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusTaskState<TYPES, I>
-                    {
-                        error!("Block payload commitment does not equal da cert payload commitment. View = {}", *view);
-                        return false;
-                    }
-                    if let Ok(vote) = QuorumVote::<TYPES>::create_signed_vote(
-                        QuorumData {
-                            leaf_commit: proposed_leaf.commit(),
-                        },
-                        view,
-                        &self.public_key,
-                        &self.private_key,
-                    ) {
-                        GeneralConsensusMessage::<TYPES>::Vote(vote)
-                    } else {
-                        error!("Unable to sign quorum vote!");
-                        return false;
-                    }
-                } else {
-                    error!(
-                        "Invalid DAC in proposal! Skipping proposal. {:?} cur view is: {:?}",
-                        cert, self.cur_view
-                    );
-                    return false;
-                };
-
-                if let GeneralConsensusMessage::Vote(vote) = message {
-                    debug!(
-                        "Sending vote to next quorum leader {:?}",
-                        vote.get_view_number() + 1
-                    );
-                    // Add to the storage that we have received the VID disperse for a specific view
-                    if let Some(vid_share) = vid_shares.get(&self.public_key) {
-                        if let Err(e) = self.storage.write().await.append_vid(vid_share).await {
-                            error!(
-                                "Failed to store VID Disperse Proposal with error {:?}, aborting vote",
-                                e
-                            );
-                            return false;
-                        }
-                    } else {
-                        error!("Did not get a VID share for our public key, aborting vote");
-                        return false;
-                    }
-                    broadcast_event(Arc::new(HotShotEvent::QuorumVoteSend(vote)), event_stream)
-                        .await;
-                    return true;
-                }
-            }
-            debug!(
-                "Received VID share, but couldn't find DAC cert for view {:?}",
-                *proposal.get_view_number(),
-            );
-            return false;
-        }
-        debug!(
-            "Could not vote because we don't have a proposal yet for view {}",
-            *self.cur_view
-        );
+    /// Ignores old vote behavior and lets `QuorumVoteTask` take over.
+    #[cfg(feature = "dependency-tasks")]
+    async fn vote_if_able(&mut self, _event_stream: &Sender<Arc<HotShotEvent<TYPES>>>) -> bool {
         false
     }
 
-    /// Validate the VID disperse is correctly signed and has the correct share.
+    /// Validates whether the VID Dispersal Proposal is correctly signed
```

Comment:
Nit: This seems to be the old documentation before I added the VID share validation. It can be updated after syncing with main.